### PR TITLE
Draft of 3D analysis section

### PR DESCRIPTION
**Pull request by LauraOlivera**
I've started the draft of the 3D analysis section. I would say the bulk of the text is mostly there, however, I haven't included any figures yet, mainly because I think there are several options and I wanted some feedback before I put the work into getting the figures into the paper. 

As mentioned in the text, I simulated 3 sources with different morphology and spectra. Then I fitted models to the simulated dataset and recovered the simulated parameters pretty closely. 

My current proposal for plots to add (perhaps as a 3 panel figure, or 2 panel with the maps + a different figure for the histogram) is:
* Significance map 
image

* Residual map after fitting
image

* Contribution to a region around the point source for each model
image

The colorbars would have labels and I would think the choice of colors for the last plot a bit better.

Questions:
- Which regions do I mark in the maps? in the screenshots above I went for a "everything approach", meaning we have in black the simulated spatial models, in red the ones that result from the fit and in blue the region selected for the third figure. This is obviously too much. My proposal would be at the very minimum keep the region represented in blue, and maybe one of the other two sets. But I am open to suggestions.
- What about spectral residuals? The residual map could also have a couple of energy bins and be a `plot_grid()` instead, but that would take up more space.
image

- There is also the comparison of spectral fit between simulation and result of the fit. This would go more into the simulation and less into the 3D analysis I guess, but I am not sure how clear it is that you also get spectra from the 3D analysis from the current setup.
image
- Flux points?

But it already feels like there are too many plots, and so I wanted to get some feedback on which ones to keep, and maybe there are better ways to represent the result than the ones I came up with... So suggestions welcome!

### CODE REVIEW

File: src/text/3-applications-subsections/3D-analysis.tex [hunk 1]
```diff
@@ -1,12 +1,26 @@
 \subsection{3D Analysis}
 \label{ssec:3d-analysis}
-
 \todo{Laura}
+The 1D analysis approach is a powerful tool to measure the spectrum of an isolated source. However, more complicated situations, such as field of views containing several overlapping sources, require a more careful treatment. Because it is based on integration over a region, the 1D approach can't disentangle the contribution of each source to the total flux in the selected region. Similarly, sources with extended or complex morphology can result in the measured flux being underestimated, and heavily dependent on the choice of extraction region.
```
Comment (AtreyeeS) [inline src/text/3-applications-subsections/3D-analysis.tex:4]: 3D analysis also has some more advantages over the 1D style (eg: taking energy dependent psf into account, for studies of very large sources covering the full FoV, etc). I guess you can lessen the comparison with the 1D case, and mention a 3D likelihood as performed, eg, in Fermi-LAT (and refer to Lars' validation paper).

File: src/text/3-applications-subsections/3D-analysis.tex [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+
+For such situations a more complex approach is needed, the so-called 3D analysis. The three relevant dimensions are the two spatial angular coordinates and an energy axis. In this framework, a combined spatial and spectral model (that is, a SkyModel, see Section~\ref{ssec:gammapy-modeling}) is fitted to the sky-maps that were previously derived from the data and bundled into a MapDataset (see Sections~\ref{ssec:gammapy-makers,ssec:gammapy-datasets}).
+
+Starting from the \irfs corresponding to the same three simulated \cta observations used in Section~\ref{ssec:1d-analysis}, we can create a MapDataset via the MapDatasetMaker. However, we will not use the simulated event lists provided by \cta but instead, use the method MapDataset.fake() to simulate measured counts from the combination of several SkyModel instances. In particular we simulate:
```
Comment (AtreyeeS) [inline src/text/3-applications-subsections/3D-analysis.tex:8]: Do all the 3 observations have the same IRF?

Comment (AtreyeeS) [reply src/text/3-applications-subsections/3D-analysis.tex:8]: Maybe mention that a `DL4` dataset is directly simulated

File: src/text/3-applications-subsections/3D-analysis.tex [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
+	\item An extended source with Gaussian morphology located at (l=0.4\textdegree, b=0.15\textdegree) with $\sigma$=0.2\textdegree and a log-parabola spectral shape.
+	\item A large shell-like structure centered around (l=0.06\textdegree, b=0.6\textdegree) with a radius and width of 0.6\textdegree and 0.3\textdegree respectively and a power-law spectral shape.
+\end{enumerate}
+The position and sizes of the sources have been selected so that their contributions overlap. This can be clearly seen in the significance map shown in the left panel of Figure \todo{X}. This map was produced with the ExcessMapEstimator (see Section~\ref{ssec:gammapy-estimators}) with a correlation radius of 0.1\textdegree.
+
+We can now fit the same model shapes to the simulated data and retrieve the best-fit parameters, which are in all cases well within 1$\sigma$ of the simulated parameters. We can now compute the residual significance map after removing the contribution from each model. This is done again via the ExcessMapEstimator. As can be seen in Figure \todo{X}, there are no regions above or below 5$\sigma$.
```
Comment (AtreyeeS) [inline src/text/3-applications-subsections/3D-analysis.tex:16]: Do you want to talk of agreement with the simulated parameters?

**Review by AtreyeeS — COMMENTED** [review]
Thanks a lot @LauraOlivera ! This section really displays the most general use of gammapy, and I think it is very well written already. Some general points regarding what you brought up

- Show the significance map with the spatial regions (simulated parameters) plotted on top? Instead of plotting the on the flattened significance map

- For me, the spectral residuals is more of diagnostics rather than something as a final result. 

- Not show any comparison for the spectral fits, but plot the flux points (with the spectral butterfly) for the 3 sources. (this can be done on one plot I guess). You can mention that the flux points can be obtained independently for each model component after doing the fit.

-

**Comment by adonath** [comment]
For reference some comments from the meeting on Monday:
- Please propose a single figure with 3 axes using either intermediate or two column width
- Make the background a bit more visible by either lowering the flux of the source, or choosing a non-linear colormap stretch, to enhance weaker structures
- Let me know if you need help with the snakemake setup for the data download / data reduction

File: src/text/3-applications-subsections/3D-analysis.tex (hunk 2 above)
Comment (LauraOlivera) [reply src/text/3-applications-subsections/3D-analysis.tex:8]: about the IRFs: that's a good point, I think that is the case. But I am not 100% sure, also I don't think anything would change in the text if so. Or what do you propose?

File: src/text/3-applications-subsections/3D-analysis.tex (hunk 1 above)
Comment (LauraOlivera) [reply src/text/3-applications-subsections/3D-analysis.tex:4]: Very good points. I thought the 1D comparison would be a good way to connect this section to the previous one, but maybe it is a bit limiting... I'll rethink it and add a citation to Lars' paper. Thanks!!

File: src/text/3-applications-subsections/3D-analysis.tex (hunk 3 above)
Comment (LauraOlivera) [reply src/text/3-applications-subsections/3D-analysis.tex]: I removed that sentence

**Comment by LauraOlivera** [comment]
I made a three panel figure, which I think is not too small. I had to sacrifice the labels in the colorbars (that would say "Significance" in both left and middle images) and instead it is stated in the caption what the maps are. I hope this is OK, I think that including the labels made everything too crowded and small. If they need to be included I would suggest splitting the figure into two (one with the maps and another one with the histograms).

[Here](https://mcloud.mpi-hd.mpg.de/index.php/s/xK9AkXQLEamdgST) you can find the draft with those changes, including the figure (Fig 14). I am not completely satisfied with the text yet, as I've only partially included some of @AtreyeeS suggestions. Will keep working on that...

**Comment by adonath** [comment]
Thanks @LauraOlivera, I'd be happy to merge the PR as is for now. Do you have any more comments @AtreyeeS?

**Comment by AtreyeeS** [comment]
I think we can merge now and follow up later

**Comment by LauraOlivera** [comment]
Ready to merge on my side :+1:

**Comment by adonath** [comment]
Thanks @LauraOlivera!

**Pull request merged**